pilot/pkg/serviceregistry/kube/controller/controller_test.go

```diff
@@ -890,6 +896,436 @@ func TestController_Service(t *testing.T) {
 	}
 }
 
```

Comment:
We have tests calling get/list of stuff, but nothing triggering the event handler. the code is pretty trivial anyways but could be nice to cover

Reply:
when I run with coverage, [this test block](https://github.com/istio/istio/pull/29802/files/d4801ef332a659ad9ca624dce284302c2765c918#diff-e1ab47ca99afd12c96075f5ff81859680711551973e1c9c8eee7e3d73493b513R1021-R1026) covers [this block](https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/controller.go#L510-L523), which in turn covers the [filter's event handler wrapper](https://github.com/istio/istio/pull/29802/files#diff-9b1e89a7f445a55fef998ca65bf562bab5f072b96af6b5326efce292eec2f72bR49-R55)